### PR TITLE
Emails redesign

### PR DESCRIPTION
**Pull request by Developing-Gamer**
## Summary by CodeRabbit

* **New Features**
  * WYSIWYG inline editing (commit/preview) for email templates, themes, and drafts, plus a mock-AI edit mode.
  * Delete actions for email drafts, templates, and themes.
  * Device viewport previews with visible editable regions.

* **Improvements**
  * Richer editor layout (save/undo, confirm, debug), tighter preview/editor integration, in-iframe editing tooling, and redesigned glass-card email management UI.
  * Thread/chat and editor UX polish; improved tooltips and layout behavior.

* **Tests**
  * New unit and end-to-end tests for templates, drafts, previews, and WYSIWYG edits.

* **Documentation**
  * Added dashboard design guide section for catalog & product card grid.

---

> [!NOTE]
> Modernizes email authoring and AI integration across backend and dashboard.
> 
> - Switches AI to OpenRouter (`STACK_OPENROUTER_API_KEY`), updates internal chat to OpenRouter with mock mode fallback
> - Adds `POST /internal/wysiwyg-edit` AI endpoint to apply WYSIWYG text edits to JSX sources
> - Extends `POST /emails/render-email` to accept `editable_markers`/`editable_source` and return `editable_regions`
> - Adds delete routes for email drafts (`DELETE /internal/email-drafts/:id`), templates (`DELETE /internal/email-templates/:templateId`), and themes (`DELETE /internal/email-themes/:id`) with safeguards (no default/active theme deletion)
> - Improves template/theme management: validates `theme_id`, uses `defaultNewTemplateSource`, precise config overrides
> - Redesigns dashboard email editors with viewport selector (phone/tablet/desktop), AI WYSIWYG commit, save/undo, unsaved-change guard, and shared-SMTP warnings
> - Updates list pages to support create/delete flows and polished glassmorphic UI; adds Design Language page and catalog/grid guide
> - Fixes globe rendering (depth/culling, hover) and chart tooltip layering/cursor behavior
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9ae5ad7efa0028202125036a08562c459de92aa6. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx [hunk 1]
```diff
@@ -85,6 +104,16 @@ export const POST = createSmartRouteHandler({
...
     };
+    } catch (error) {
+      // Log the error for debugging
+      console.error("AI chat generation error:", error);
+
+      // Re-throw as a user-friendly error
+      throw new StatusError(
+        StatusError.InternalServerError,
+        `Failed to generate AI response: ${error instanceof Error ? error.message : 'Unknown error'}`
+      );
```
Comment (N2D4) [inline apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx]: bad

File: packages/template/src/lib/stack-app/apps/implementations/admin-app-impl.ts [hunk 2]
```diff
@@ -504,9 +504,31 @@ export class _StackAdminAppImplIncomplete<HasTokenStore extends boolean, Project
...
+      const current = this._adminEmailTemplatesCache.getIfCached([]);
+      if (current.status === "ok" && current.data.status === "ok") {
+        if (current.data.data.some(t => t.id === result.id)) {
+          break;
+        }
+      }
+      await new Promise(resolve => setTimeout(resolve, 200));
+      await this._adminEmailTemplatesCache.refresh([]);
+    }
+
```
Comment (N2D4) [inline packages/template/src/lib/stack-app/apps/implementations/admin-app-impl.ts]: this is a bug we should find out why it happens and fix it — instead of a workaround

File: packages/template/src/lib/stack-app/apps/implementations/admin-app-impl.ts [hunk 3]
```diff
@@ -504,9 +504,31 @@ export class _StackAdminAppImplIncomplete<HasTokenStore extends boolean, Project
...
     return result;
   }
 
+  async deleteEmailTemplate(id: string): Promise<void> {
+    await this._interface.deleteEmailTemplate(id);
+    const current = this._adminEmailTemplatesCache.getIfCached([]);
+    if (current.status === "ok" && current.data.status === "ok") {
+      this._adminEmailTemplatesCache.forceSetCachedValue([], Result.ok(current.data.data.filter((t) => t.id !== id)));
+    }
```
Comment (N2D4) [inline packages/template/src/lib/stack-app/apps/implementations/admin-app-impl.ts]: as above (no workarounds)

File: apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx (hunk 1 above)
Comment (Developing-Gamer) [reply apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx]: Resolved

File: packages/template/src/lib/stack-app/apps/implementations/admin-app-impl.ts (hunk 2 above)
Comment (Developing-Gamer) [reply packages/template/src/lib/stack-app/apps/implementations/admin-app-impl.ts]: Couldn't reproduce the bug I was having previously, could be a package/cursor browser cache issue. Removed the workaround, everything is working fine without it as well.

File: packages/template/src/lib/stack-app/apps/implementations/admin-app-impl.ts (hunk 3 above)
Comment (Developing-Gamer) [reply packages/template/src/lib/stack-app/apps/implementations/admin-app-impl.ts]: same as above (no workaround needed anymore)

File: apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx [hunk 4]
```diff
@@ -22,7 +23,19 @@ const toolCallContentSchema = yupObject({
...
+  content: yupMixed().defined(),
+});
+
+const aiProvider = getEnvVariable("STACK_AI_PROVIDER", "openai");
+const openai = createOpenAI({
+  apiKey: aiProvider === "openrouter"
+    ? getEnvVariable("STACK_OPENROUTER_API_KEY", "MISSING_OPENROUTER_API_KEY")
+    : getEnvVariable("STACK_OPENAI_API_KEY", "MISSING_OPENAI_API_KEY"),
+  baseURL: aiProvider === "openrouter" ? "https://openrouter.ai/api/v1" : undefined,
+});
```
[the remark this answers is not in the document]
Comment (N2D4) [reply apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx]: reopening this, coderabbit is right — don't pass a default argument so the errors are better

File: apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx [hunk 5]
```diff
@@ -53,10 +62,18 @@ export const POST = createSmartRouteHandler({
   }),
   async handler({ body, params, auth: { tenancy } }) {
     const adapter = getChatAdapter(body.context_type, tenancy, params.threadId);
+    const modelName = getEnvVariable("STACK_AI_MODEL", getEnvVariable("STACK_OPENAI_MODEL", "gpt-4o"));
```
Comment (N2D4) [inline apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx]: gpt-4o seems like a bad default, we also don't want a default here i think

File: apps/backend/src/app/api/latest/internal/email-drafts/[id]/route.tsx [hunk 6]
```diff
@@ -78,3 +78,30 @@ export const PATCH = createSmartRouteHandler({
...
+    await prisma.emailDraft.delete({
+      where: { tenancyId_id: { tenancyId: tenancy.id, id: params.id } },
+    });
+    return {
+      statusCode: 200,
+      bodyType: "json",
+      body: { ok: "ok" },
+    };
+  },
+});
```
[the remark this answers is not in the document]
Comment (N2D4) [reply apps/backend/src/app/api/latest/internal/email-drafts/[id]/route.tsx:114]: reopening this as well, tests for this are missing

File: apps/backend/src/app/api/latest/internal/email-templates/[templateId]/route.tsx [hunk 7]
```diff
@@ -38,6 +38,12 @@ export const PATCH = createSmartRouteHandler({
     if (!Object.keys(templateList).includes(templateId)) {
       throw new StatusError(StatusError.NotFound, "No template found with given id");
     }
+
+    // Validate theme_id to prevent object injection
+    if (body.theme_id !== undefined && body.theme_id !== null && body.theme_id !== false && typeof body.theme_id !== 'string') {
+      throw new StatusError(StatusError.BadRequest, "theme_id must be a string, null, or false");
+    }
```
Comment (N2D4) [inline apps/backend/src/app/api/latest/internal/email-templates/[templateId]/route.tsx]: why isn't this on the yup schema above? if it is, why is it here again?

File: packages/stack-ui/src/components/data-table/column-header.tsx
Comment (N2D4) [inline packages/stack-ui/src/components/data-table/column-header.tsx:1]: these changes in the wrong file? should probably be in apps/dashboard instead (whcih has a file with the same name)?

File: packages/stack-ui/src/components/data-table/view-options.tsx
Comment (N2D4) [inline packages/stack-ui/src/components/data-table/view-options.tsx:1]: as above

File: apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/email-drafts/[draftId]/page-client.tsx [hunk 8]
```diff
@@ -38,6 +39,19 @@ export default function PageClient({ draftId }: { draftId: string }) {
...
+  const handleSave = async () => {
+    try {
+      await stackAdminApp.updateEmailDraft(draftId, { tsxSource: currentCode, themeId: selectedThemeId });
+      toast({ title: "Draft saved", variant: "success" });
+    } catch (error) {
+      if (error instanceof KnownErrors.EmailRenderingError) {
+        toast({ title: "Failed to save draft", variant: "destructive", description: error.message });
+        return;
+      }
+      toast({ title: "Failed to save draft", variant: "destructive", description: "Unknown error" });
```
Comment (N2D4) [inline apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/email-drafts/[draftId]/page-client.tsx]: don't use toasts, toasts are bad (also true everywhere else)

File: apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx
Comment (N2D4) [inline apps/backend/src/app/api/latest/internal/ai-chat/[threadId]/route.tsx:1]: the draft email editor doesn't work. i think it only works for themes, not drafts and themes because it incorrectly generates stuff like \<Head> and \<Html>. the setup is something like this:  
  


```
<Theme>
  <Template>
    <Draft />
  </Template>
</Theme>

```

File: apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/email-drafts/[draftId]/page-client.tsx
Comment (N2D4) [inline apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/email-drafts/[draftId]/page-client.tsx:1]: certain runtime errors, such as invalid layout (eg. \<Html> anywhere, or `\<Subject value={anything that's not a string} >`) or syntax errors are not surfaced correctly. please be careful about the error handling, make sure you interface all those errors consistently and correctly (no matter where they happen), and don't use toasts for them

File: packages/stack-shared/src/helpers/emails.ts [hunk 9]
```diff
@@ -42,24 +81,25 @@ export const emptyEmailTheme = deindent`
...
   return (
     <Html>
       <Head />
       <Tailwind>
-        <Body className="bg-[#fafbfb] font-sans text-base">
-          <Container className="bg-white p-[45px] rounded-lg">
-            <ProjectLogo data={projectLogos} mode="light" />
-            {children}
-          </Container>
+        <Body className="bg-[#fafbfb] font-sans text-base m-0 p-0" style={{ overflowX: 'hidden' }}>
```
Comment (N2D4) [inline packages/stack-shared/src/helpers/emails.ts:92]: is there a reason for this change?

**Review by N2D4 — CHANGES_REQUESTED** [review]
really really really good work! some amazing stuff in here. visuals are great! most comments are around error handling — make sure to really think about the edge cases because it's those that make a product good  
  
also, make sure you really thoroughly test everything you put into reviews. there were a few errors that were really severe (eg. updating any code in the code editor woudl throw an application error), so make sure you test all functionalities

File: apps/backend/src/app/api/latest/internal/email-drafts/[id]/route.tsx [hunk 10]
```diff
@@ -25,7 +27,15 @@ export const GET = createSmartRouteHandler({
...
-    const d = await prisma.emailDraft.findFirstOrThrow({ where: { tenancyId: tenancy.id, id: params.id } });
+    let d;
+    try {
+      d = await prisma.emailDraft.findFirstOrThrow({ where: { tenancyId: tenancy.id, id: params.id } });
+    } catch (error) {
+      if (error instanceof Prisma.PrismaClientKnownRequestError && error.code === "P2025") {
+        throw new StatusError(StatusError.NotFound, "No draft found with given id");
+      }
+      throw error;
+    }
```
[the remark this answers is not in the document]
Comment (N2D4) [reply apps/backend/src/app/api/latest/internal/email-drafts/[id]/route.tsx]: incorrectly resolved, coderabbit is correct
